lib/rules/no-unsupported-features/es-syntax.js

```diff
@@ -378,7 +378,7 @@ const features = {
         ruleId: "no-dynamic-import",
         cases: [
             {
-                supported: null,
+                supported: new Range(">=12.17 <13 || >=13.2"),
```

Comment:
I thought through several ways of accomplishing this, and landed on allowing `case.supported` to be a `Range` instance. Here are my reasons:

* When it's a string, the `getSemverRange` function is called from `isNotSupportingVersion` while automatically prepending a `<` character, to make it a negative assertion to match the `Not` in the function name. I can't change `getSemverRange` because it's nicely memoized and used all over the place in different ways.
* I thought of making supported an array, but then exactly _how_ to combine the array values wasn't clear.
* Given those things, I figured that we could just say, "If you want to specify some highly-specific range, just go ahead and make a `Range` and we'll use that."